lib/instrumentation/span.js

```diff
@@ -46,6 +47,13 @@ function Span (transaction, name, type, opts) {
   this._agent.logger.debug('start span %o', { span: this.id, parent: this.parentId, trace: this.traceId, name: name, type: type })
 }
 
+Span.prototype.toString = function () {
```

Comment:
This is enforcing a particular format which may be out of place with the user's logging. That's the main reason why I wasn't too keen on adding integration with unstructured loggers in the first place.

What about this as an alternative: add an `Agent.currentTraceIds` method, similar to `Agent.currentTraceparent`, which will return an object containing the IDs of the current span if there is one, or otherwise the current transaction if there is one.

Then users could use it like `querystring.stringify(apm.currentTraceIds, ',', '=')`, and it would also be used for structured logging, e.g. labels in winston.

WDYT?

Reply:
We already have getters for the various ids, so that doesn't really seem all that useful to me. If the format generated by toString() doesn't produce what you want, you can always use those directly when writing your own log integration. 🤷‍♂

Reply:
You have to have the relevant object (span or transaction) before you can get its IDs, and there's some complexity here:

 - there may or may not be a transaction
 - there may or may not be a span
 - the agent currently doesn't produce no-op spans for non-sampled transactions, but that may not be the case forever; if we start creating no-op spans, then it wouldn't make sense to use the span ID

So without something like `currentTraceIds`, how would you get the labels for for the current span or transaction for a structured logger? I imagine something like the following, assuming no-op spans in the future:

```js
const span = apm.currentSpan
if (span && span.sampled) {
    return {'trace.id': span.traceId, 'span.id': span.id}
} else {
    const tx = apm.currentTransaction
    return tx ? {'trace.id': tx.traceId, 'transaction.id': tx.id} : {}
}
```

This is relevant for unstructured loggers too. They also need to call `toString` on the current-sampled-span-or-transaction, so the conditional parts of of that code still apply. Then if people want to format it differently, everyone needs to repeat that.

Maybe I'm missing a simpler way?

Reply:
Fair point about having to deal with possible non-existence of spans/transactions. It puts extra work on the user for serialization though. Could maybe go for a hybrid approach, something like:

```js
const { stringify } = require('querystring')

class Ids {
  toString () {
    return stringify(this)
  }
}

class SpanIds extends Ids {
  constructor(span) {
    super()
    this['span.id'] = span.id
    this['trace.id'] = span.traceId
  }
}

class TransactionIds extends Ids {
  constructor(transaction) {
    super()
    this['transaction.id'] = transaction.id
    this['trace.id'] = transaction.traceId
  }
}

function currentTraceIds () {
  const span = apm.currentSpan
  if (span && span.sampled) {
      return new SpanIds(span)
  } else {
      const tx = apm.currentTransaction
      return tx ? new TransactionIds(tx) : new Ids()
  }
}
```

Reply:
That seems like a good approach to me. Then people who want to override the string format could either change `Ids.prototype.toString` or pass the object into a function depending on their use case.

BTW is it typical to use a querystring format in Node.js app logs? I would have thought space-separated would be more common, but I don't know.

Reply:
I just pushed an update with a new API based on your feedback. Let me know what you think.

Reply:
The new API looks great to me, thanks. Good idea adding `.ids` to Transaction and Span too.